source/extensions/filters/http/aws_request_signing/aws_request_signing_filter.cc

```diff
@@ -80,19 +85,28 @@ Http::FilterDataStatus Filter::decodeData(Buffer::Instance& data, bool end_strea
   try {
     ENVOY_LOG(debug, "aws request signing from decodeData");
     ASSERT(request_headers_ != nullptr);
-    config_->signer().sign(*request_headers_, hash);
-    config_->stats().signing_added_.inc();
-    config_->stats().payload_signing_added_.inc();
+    config.signer().sign(*request_headers_, hash);
+    config.stats().signing_added_.inc();
+    config.stats().payload_signing_added_.inc();
   } catch (const EnvoyException& e) {
     // TODO: sign should not throw to avoid exceptions in the request path
     ENVOY_LOG(debug, "signing failed: {}", e.what());
-    config_->stats().signing_failed_.inc();
-    config_->stats().payload_signing_failed_.inc();
+    config.stats().signing_failed_.inc();
+    config.stats().payload_signing_failed_.inc();
   }
 
   return Http::FilterDataStatus::Continue;
 }
 
+FilterConfig& Filter::getConfig() const {
+  auto* config = const_cast<FilterConfig*>(
```

Comment:
Why this need to be a non-const?

Reply:
Hey lizan,
It needs to be non-const because after getting the config using above method we have calls to non-const member methods on config such as [.stats()](https://github.com/envoyproxy/envoy/blob/3914a92fb2d086d450dcf4cc3f0d2571a52c60f0/source/extensions/filters/http/aws_request_signing/aws_request_signing_filter.cc#L57) and .signer().